### PR TITLE
Fix memory leak in unit tests - improve clean up in tearDown method

### PR DESCRIPTION
**Pull request by bancer**
There is a similar clean up in 2.x - https://github.com/cakephp/cakephp/blob/2.x/lib/Cake/TestSuite/CakeTestCase.php#L168

### CODE REVIEW

File: src/TestSuite/TestCase.php [hunk 1]
```diff
@@ -161,6 +161,7 @@ public function tearDown()
             Configure::write($this->_configure);
         }
         $this->getTableLocator()->clear();
+        unset($this->_configure, $this->_pathRestore, $this->_tableLocator);
```
Comment (ADmad) [inline src/TestSuite/TestCase.php]: Unsetting declared properties will make static analyzers unhappy. Set them to null / initial declared value instead.

Comment (bancer) [reply src/TestSuite/TestCase.php]: Setting to `null` or an empty array does not free all the memory though.

File: src/ORM/Locator/LocatorAwareTrait.php [hunk 2]
```diff
@@ -70,7 +70,7 @@ public function setTableLocator(LocatorInterface $tableLocator)
      */
     public function getTableLocator()
     {
-        if (!$this->_tableLocator) {
+        if (!isset($this->_tableLocator)) {
```
Comment (ADmad) [inline src/ORM/Locator/LocatorAwareTrait.php]: This should be reverted or changed to `$this->_tableLocator === null`

**Comment by dereuromark** [comment]
Nice work :+1:

File: src/ORM/Locator/LocatorAwareTrait.php (hunk 2 above)
Comment (bancer) [reply src/ORM/Locator/LocatorAwareTrait.php]: @ADmad, @dereuromark The problem with `$this->_tableLocator === null` and `!$this->_tableLocator` is that it results in: 
```
Undefined property: MyTest::$_tableLocator

.../Vendor/cakephp/orm/Locator/LocatorAwareTrait.php:73
.../src/TestSuite/MyTestCase.php:59
```

**Pull request merged**